File: cs50x/psets/1/cash/cash.c
#include <stdio.h>
#include <math.h>
#include <cs50.h>

int whileCents(int n, int m);

int main(void)
{
    float dollars;
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);

    int cents = round(dollars * 100);
    int nCoins = 0;
    int coins[4] = {25, 10, 5, 1};

    for (int i = 0; i < 4; i++)
    {
        nCoins += whileCents(cents, coins[i]);
        cents %= coins[i];
    }

    printf("%i\n", nCoins);
}

int whileCents(int n, int m)
{
    int result = 0;
    while (n >= m)
    {
        n -= m;
        result++;
    }
    return result;
}

File: cs50x/psets/6/cash/cash.py
from cs50 import get_float


def main():
    while True:
        dollars = get_float("Change owed: ")
        if dollars > 0:
            break

    cents = round(dollars * 100)
    nCoins = 0
    coins = [25, 10, 5, 1]

    for coin in coins:
        nCoins += whileCents(cents, coin)
        cents %= coin

    print(f"{nCoins}")


def whileCents(n, m):
    result = 0
    while n >= m:
        n -= m
        result += 1
    return result


main()